packages/components/src/components/input-field/input-field.tsx

```diff
@@ -1,12 +1,70 @@
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
 import React from 'react';
 import { isCryptocurrency, getCurrencyDisplayCode } from '@deriv/shared';
-import IncrementButtons from './increment-buttons.jsx';
-import Input from './input.jsx';
+import IncrementButtons from './increment-buttons';
+import Input from './input';
 import Tooltip from '../tooltip';
 import Text from '../text';
 
+export type TChangeEvent = { target: { name: string; value: number | string } };
+
+// ToDo: Refactor input_field
+// supports more than two different types of 'value' as a prop.
+// Quick Solution - Pass two different props to input field.
+type TInputFieldProps = {
+    ariaLabel: string;
+    checked?: number | string;
+    className?: string;
+    classNameDynamicSuffix?: string;
+    classNameInlinePrefix?: string;
+    classNameInput?: string;
+    classNamePrefix?: string;
+    classNameWrapper?: string; // CSS class for the component wrapper
+    currency: string;
+    current_focus: string;
+    data_testid?: string;
+    data_tip?: string;
+    data_value?: string;
+    decimal_point_change?: number; // Specify which decimal point must be updated when the increment/decrement button is pressed
+    error_message_alignment?: string;
+    error_messages?: string[];
+    format?: (new_value?: string) => string;
+    fractional_digits: number;
+    helper?: string;
+    icon?: React.ElementType;
+    id?: string;
+    increment_button_type?: 'button' | 'submit' | 'reset';
```

Comment:
mauby it would be better to create separate type and reuse it?
I mean for increment_button_type and inputmode

Reply:
@yauheni-deriv,   `increment_button_type` and `inputmode` are actually different. Please See below for clarity

`inputmode?: 'search' | 'text' | 'none' | 'tel' | 'url' | 'email' | 'numeric' | 'decimal';`

`increment_button_type?: 'button' | 'submit' | 'reset';`

`inputmode` is for `HTMLInputElement` and `increment_button_type` is for `HTMLButtonElement`

Please look into this again

Reply:
I see, I meant, that you can use twice in your code, and increment_button_type also. it could be two different reusable types

Reply:
it could be used here and in increment-buttons.tsx

Reply:
Sure @yauheni-deriv